### DISCUSSION TITLE
coloring, filtering, reading properties of Tile3DLayer

### DISCUSSION THREAD

**Discussion by enersis-pst** [visgl/deck.gl#5493] [General]
Hello,

i start to load b3dm tiles into mapbox by using deck.gl.
This works. But now i want to to do some filtering, dynamic coloring of the object probierty and also get the properties at onHover, onClick. Do someone could explain me how?

getColor, getPickingInfo woudnt be called and event have a huge object but i coudnt find any properties, which should be included.

```
const myDeckLayer =  new MapboxLayer({
  id: 'tile-3d-layer',
  type: Tile3DLayer,
  data: config.url,
  loader: Tiles3DLoader,
  pickable: true,
  autoHighlight: true,
  onHover: (info, event) => console.log('Hovered:', info, event),
  onClick: (info, event) => console.log('Clicked:', info, event),
  getPickingInfo: (pickParams) => console.log("PickInfo", pickParams),
  getColor: (ob) => {console.log("color", ob); return [255,0,0,255];}
});


map.on('load', () => {
  map.addLayer(myDeckLayer, "test");
});
```

**Comment by ibgreen** [comment]
We are currently working on supporting object selection for I3S tilesets. This will not directly add support for 3D Tiles but will put a lot of the necessary plumbing in place.

The work is fairly involved, but if you have the skills and the appetite you are welcome to join the effort.

**Reply by ibgreen** [reply]
Also when it comes to styling we haven't made up our minds. There was some unfinished work on supporting the 3D Tiles styling spec, but we likely want something that supports both 3D Tiles and I3s.

**Reply by enersis-pst** [reply]
Thanks for that info. Yes, if it could fit, it would be nice.
Is it possible to see whats the roadmap? 
It would be nice to work on that stuff. Maybe i get some time for it, but then it wouldnt be so easy for me to get into the code ;-)

**Reply by ibgreen** [reply]
The active work is in loaders.gl. For now that would be the place to engage. There is some information in the trackers in the issues there.

At this stage we are mainly collecting requirements. Opening a detailed issue there could be a way to help.
